tests/Http/Controller/PtpControllerTest.php

```diff
@@ -0,0 +1,147 @@
+        $image = Image::factory()->create(['volume_id' => $this->volume()->id]);
+
+        $imageAnnotation = ImageAnnotation::factory()->create([
+            'image_id' => $image->id,
+            'shape_id' => Shape::pointId(),
+        ]);
+
+        $url = '/api/v1/send-ptp-job/'.$this->volume()->id;
+
+        $this->beEditor();
+        $this->postJson($url)->assertStatus(200);
+
+        //$this->volume does not update with attrs
+        $volume = Volume::where('id', $this->volume()->id)->first();
+
+        $this->assertTrue(isset($volume->attrs['ptp_job_id']));
+
+        //If a ptp_job_id is set on a volume, we should get an error
+        $this->postJson($url)->assertStatus(400)->assertJson(
+            fn (AssertableJson $json) =>
+                $json->where('message', 'Another point to polygon conversion job is running in this volume!')
+                     ->etc()
+        );
+    }
+
+    public function testVideoVolumes()
+    {
+        $this->volume(['media_type_id' => MediaType::videoId()]);
+
+        $this->beEditor();
+
+        $url = '/api/v1/send-ptp-job/'.$this->volume()->id;
+
+        $this->postJson($url)->assertStatus(400)->assertJson(
+            fn (AssertableJson $json) =>
+                $json->where('message', 'Point to polygon conversion cannot be executed on this volume!')
+                     ->etc()
+        );
+    }
+
+    public function testTiledImages()
+    {
+
+        $image = Image::factory()->create([
+            'volume_id' => $this->volume()->id,
+            'tiled' => true,
+        ]);
+
+        $imageAnnotation = ImageAnnotation::factory()->create([
+            'image_id' => $image->id,
+            'shape_id' => Shape::pointId(),
+        ]);
+
+        $this->beEditor();
+
+
+        $url = '/api/v1/send-ptp-job/'.$this->volume()->id;
+
+        $this->postJson($url)->assertStatus(400)->assertJson(
+            fn (AssertableJson $json) =>
+                $json->where('message', 'Point to polygon conversion cannot be executed on this volume!')
+                     ->etc()
+        );
+    }
+
+    public function testNoImageAnnotations()
+    {
+        $this->beEditor();
+
+        $url = '/api/v1/send-ptp-job/'.$this->volume()->id;
+
+        $this->postJson($url)->assertStatus(400)->assertJson(
+            fn (AssertableJson $json) =>
+                $json->where('message', 'No point annotations to convert!')
+                     ->etc()
+        );
+    }
+
+    public function testBadJobGenerated()
+    {
+        //Test that if a bad job is generated a job ID is not present
+        $image = Image::factory()->create(['volume_id' => $this->volume()->id]);
+
+        $imageAnnotation = ImageAnnotation::factory()->create([
+            'image_id' => $image->id,
+            'shape_id' => Shape::pointId(),
+        ]);
+        $this->beEditor();
+        config(['ptp.ptp_fail_construct' => true]);
+        config(['ptp.ptp_storage_disk' => 'test']);
```

Comment:
Instead of the test config, you can probably use something like `Bus::shouldReceive(...)->andThrow(\Exception:.class)` (see [Laravel docs](https://laravel.com/docs/11.x/queues#testing)).